pipeline/lib/api.ml

```diff
@@ -0,0 +1,113 @@
+open Lwt.Infix
+module Utils = Cohttp_lwt_unix
+
+type t = {
+  benchmark :
+    version:int ->
+    benchmark_name:string option ->
+    test_index:int ->
+    Yojson.Safe.t ->
+    Models.Benchmark.t;
```

Comment:
Just to clarify why I'm suggesting to remove this record `t`: in ocaml it's very rare that you need a function inside a record... so you really have to justify it when you do it ^^

When reading the code, this type sends the message "I'm going to arbitrarily set this `benchmark` field with a lot of different unknown functions" (which isn't a good sign!). But here it's just an intermediate value :)

Reply:
Thanks for clarifying!